libfwupdplugin/fu-device.c

```diff
@@ -2049,6 +2057,55 @@ fu_device_set_logical_id (FuDevice *self, const gchar *logical_id)
 	g_object_notify (G_OBJECT (self), "logical-id");
 }
 
+/**
+ * fu_device_get_backend_id:
+ * @self: A #FuDevice
+ *
+ * Gets the ID set for the device as recognised by the backend. This is typically
+ * a Linux sysfs path or USB platform ID. If unset, it also falls back to the
+ * physical ID as this may be the same value.
+ *
+ * Returns: a string value, or %NULL if never set.
+ *
+ * Since: 1.5.8
+ **/
+const gchar *
+fu_device_get_backend_id (FuDevice *self)
+{
+	FuDevicePrivate *priv = GET_PRIVATE (self);
+	g_return_val_if_fail (FU_IS_DEVICE (self), NULL);
+	if (priv->backend_id != NULL)
+		return priv->backend_id;
+	return priv->physical_id;
```

Comment:
A FuDevice doesn't have to be created by a backend; in fact most are created inside the plugins -- although they do `->incorporate()` the backend_id I suppose. The fallback logic here was really just trying to avoid the common case of the physical ID also being the backend ID, which would save a few g_strdup's at startup. Maybe it's not worth it, if it's confusing things.